src/hooks/useNewBidEventFromChain.ts

```diff
@@ -75,28 +71,21 @@ export function useNewBidEventFromChain(saleId: string): UseBidEventFromChainRet
   }
 }
 
-export function useNewCommitmentEventFromChain(saleId: string): UseCommitmentEventFromChainReturns {
+export function useNewCommitmentEventFromChain(saleId: string) {
   const dispatch = useDispatch()
   const { account, library, chainId } = useWeb3React()
   const [t] = useTranslation()
-  const {
-    isLoading,
-    error,
-    bidsBySaleId: { [saleId]: { bids } = { bids: [] } },
-  } = useSelector(({ commitments }) => commitments)
 
   useEffect(() => {
     if (!account || !library || !chainId) {
       return
     }
     const fixedPriceSaleContract = FixedPriceSale__factory.connect(saleId, library)
-
-    fixedPriceSaleContract.on('NewCommitment', async (buyer, amount) => {
-      const totalCommitmentsByUserAddress =
-        bids.filter(bid => bid.user.address.toLowerCase() === buyer.toLowerCase()).length + 1
-
+    fixedPriceSaleContract.removeAllListeners('NewCommitment')
+    fixedPriceSaleContract.on('NewCommitment', async (buyer, amount, event) => {
       const commitment: GetAllBidsBySaleId_fixedPriceSale_commitments = {
-        id: saleId + '/commitments/' + buyer + '/' + totalCommitmentsByUserAddress,
+        id:
+          saleId + '/commitments/' + buyer + '/' + String(await (await library.getBlock(event.blockNumber)).timestamp),
```

Comment:
The reason I used block timestamps was that it was data coming directly from the blockchain and so its easier to verify that the update is unique as this gets called multiple times due to hooks setting up the listeners multiple times (I dont really know how to solve this as we want to use hooks multiple times).

For some reason also the hook getting data from redux store is not updating inside of this method which was what caused the original issue I think now that I look at it again this morning.

Since this is just for the UIs reference and is not saving to the graph I think using timestamps is the best option to ensure the new event is unique.